core/commands/storage.go

```diff
@@ -632,20 +637,41 @@ func payFullToEscrow(ctx context.Context, response *escrowPb.SignedSubmitContrac
 		log.Error(err)
 		return
 	}
+	//TODO talk with Jin for doing signature for every contract
 	// get escrow sig, add them to guard
 	sig := payinRes.EscrowSignature
-	for _, guardContract := range guardContracts {
+	for _, guardContract := range ss.GetGuardContracts() {
```

Comment:
just quick question. I saw you using session's guardContracts but where you set the value to session?

Reply:
if you wish to store that on session you can add your set function in storage/session.go, and place it in where you need them

Reply:
I saw in session.go L298 IncrementContract will change the ss.GuardContracts , correct me if my understanding wrong.

Reply:
ah, right, should be.